### PR TITLE
Push image multi

### PR DESCRIPTION
**Pull request by pacostas**
## Summary

This PR adds multi-architecture support to the push-image workflow. That way, when one or more architectures are on the build.oci and run.oci archives, is able to push all the included architectures to the registry.

## Testing
This PR has been tested on `pacostas/ubi-base-stack` and `pacostas/jammy-tiny-stack` repos.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: .github/workflows/push-image.yml [hunk 1]
```diff
@@ -4,54 +4,107 @@ on:
...
 env:
-  REGISTRIES_FILENAME: "registries.json"
+  REGISTRIES_FILEPATH: "registries.json"
+  STACKS_FILEPATH: "stacks/images.json"
 
 jobs:
   preparation:
     name: Preparation
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (mhdawson) [inline .github/workflows/push-image.yml:21]: Suggest we do this in separate PR as discussed, same comment below as well.

File: .github/workflows/push-image.yml [hunk 2]
```diff
@@ -4,54 +4,107 @@ on:
...
+            "create_build_image": (.create_build_image // false)
+          }]' ${{ env.STACKS_FILEPATH }} )
+        fi
+
+        ## Due to publish.sh script compatibility, in case of there is no build image
+        ## set a default value to the default build image which will always be available.
+        stacks=$(echo "$stacks" | jq 'map(if .create_build_image == false then .build_image = "build" else . end)')
+
+        ## Filter stacks array to include the minimum number of attributes
+        stacks=$(echo "$stacks" | jq 'map({ name, build_image, run_image })')
```
Comment (mhdawson) [inline .github/workflows/push-image.yml:84]: Is there any time where we expect there be stacks that don't have all of the required attributes?  Just asking to understand if maybe we should check/fail versus just filtering out?

File: .github/workflows/push-image.yml [hunk 3]
```diff
@@ -60,82 +113,113 @@ jobs:
 
   push:
     name: Push
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
```
Comment (mhdawson) [inline .github/workflows/push-image.yml:116]: I see now that it was inconsistent before. One being latest and the other be 22.04. In that context making the two at least match does make more sent to me. At this point not sure if latest or 24.04 makes the most sense based on what is done in other repos?

File: .github/workflows/push-image.yml [hunk 4]
```diff
@@ -60,82 +113,113 @@ jobs:
...
+          --build-archive "${GITHUB_WORKSPACE}/${{ matrix.stack.build_image }}.oci"  \
+          --run-archive "${GITHUB_WORKSPACE}/${{ matrix.stack.run_image }}.oci"
+
+        if [ "${{ needs.preparation.outputs.push_to_gcr }}" = "true" ]; then
+          for imageType in ${{ matrix.stack.build_image }} ${{ matrix.stack.run_image }}; do
+            echo "FROM docker.io/${DOCKERHUB_ORG}/${imageType}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" | \
+            docker buildx build -f - . \
+              --tag "gcr.io/${GCR_PROJECT}/${imageType}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" \
+              --tag "gcr.io/${GCR_PROJECT}/${imageType}-${{ needs.preparation.outputs.repo_name }}:latest" \
+              --platform linux/amd64,linux/arm64 \
```
Comment (mhdawson) [inline .github/workflows/push-image.yml:188]: As discussed explore how we can avoid hard coding, and test one repo where pushed to gcr are done.

File: .github/workflows/push-image.yml (hunk 2 above)
Comment (pacostas) [reply .github/workflows/push-image.yml:84]: I think that's impossible

File: .github/workflows/push-image.yml (hunk 3 above)
Comment (pacostas) [reply .github/workflows/push-image.yml:116]: in other repos we use 22

File: .github/workflows/push-image.yml (hunk 1 above)
Comment (pacostas) [reply .github/workflows/push-image.yml:21]: Based on the below comment i'll use 22 instead of latest

File: .github/workflows/push-image.yml (hunk 4 above)
Comment (pacostas) [reply .github/workflows/push-image.yml]: Added it but not tested pushing to GCR.

**Comment by pacostas** [comment]
Tested also on jammy base stack

File: .github/workflows/push-image.yml (hunk 2 above)
Comment (mhdawson) [reply .github/workflows/push-image.yml:84]: I guess this is not filtering out stacks, just filtering down to the attributes we need?

File: .github/workflows/push-image.yml [hunk 5]
```diff
@@ -65,77 +118,112 @@ jobs:
...
+          --build-ref "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.build_image }}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" \
+          --build-ref "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.build_image }}-${{ needs.preparation.outputs.repo_name }}:latest" \
+          --run-ref "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.run_image }}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" \
+          --run-ref "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.run_image }}-${{ needs.preparation.outputs.repo_name }}:latest" \
+          --build-archive "${GITHUB_WORKSPACE}/${{ matrix.stack.build_image }}.oci"  \
+          --run-archive "${GITHUB_WORKSPACE}/${{ matrix.stack.run_image }}.oci"
+
+          if [ "${{ needs.preparation.outputs.push_to_gcr }}" = "true" ]; then
+
+          platforms=$(docker manifest inspect "docker.io/${DOCKERHUB_ORG}/${{ matrix.stack.run_image }}-${{ needs.preparation.outputs.repo_name }}:${{ needs.preparation.outputs.tag }}" |
```
Comment (mhdawson) [inline .github/workflows/push-image.yml:184]: I assume this works even if there is no manifest, right? It must since you tested with the ubi stack which today is only a single arch.

**Comment by mhdawson** [comment]
> Tested also on jammy base stack

Does that push to ghcr ?

**Comment by pacostas** [comment]
Closing it, as I've opened a new one https://github.com/paketo-community/ubi-base-stack/pull/108 due to jam supports pushing one image per time
